### PR TITLE
Revert "add hololens2 inputsimulationprofile"

### PR DESCRIPTION
**Pull request by julenka**
Reverts microsoft/MixedRealityToolkit-Unity#6393

PR #6393  aimed to change the default behavior of the Hand Interaction example scene to turn use eye simulation. It's very good to know that it's possible to have the desired behavior in #6393 by modifying the input profile. However, one of the goals of MRTK is to have defaults that match the interaction model of HoloLens and Windows Mixed Reality. 

PR #6393 had the effect of turning on the head cursor when no hands are present, which does not match shell parity. There is also a second bug: the gaze cursor turns on if a hand is near a grabbable or touchable object, issue #6496.

Fixes: #6496

### CODE REVIEW

**Comment by julenka** [comment]
The reason tests did not catch this is because the change was to a simulation profile, not actual behavior. I think it may be good to add a script to validate that the default simulation profiles do not simulate eye tracking.

**Comment by julenka** [comment]
@cre8ivepark  FYI

**Comment by keveleigh** [comment]
It feels like this is more a bug in our interaction model state machine than a bug in the profile though, right?

If hand interaction is enabled, we shouldn't be turning the gaze cursor on just because the rays are temporarily off?

**Comment by wiwei** [comment]
> It feels like this is more a bug in our interaction model state machine than a bug in the profile though, right?
> 
> If hand interaction is enabled, we shouldn't be turning the eye gaze cursor on just because the rays are temporarily off?

Reading this PR, the original one, and this comment here (https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6320#issuecomment-545887152), I'm also a little bit confused - it feels like the intent of the original change was to make it so that the default HL2 profile shows off and tests the latest HL2 functionality. i.e. it feels weird to have head gaze by default, when we want to be pushing folks to use eye gaze (or at least, try out eye gaze!)

The fact that there is this side effect feels like a separate bug that we can fix. Whether or not that's critical determines whether or not we live with it, or roll it back and go again.

I think my other general question is with respect to eye gaze and cursors, is it a true statement that the rules controlling eye gaze visibility and hands are the same ones that control head gaze visibility and hands? If so we should just fix the underlying bug and maybe even leave this profile in because it gives developers the new eye tracking features on by default if they are targetting HL2 development

**Comment by gilbdev** [comment]
> PR #6393 had the effect of turning on the head cursor when no hands are present

> it feels like the intent of the original change was to make it so that the default HL2 profile shows off and tests the latest HL2 functionality. i.e. it feels weird to have head gaze by default, when we want to be pushing folks to use eye gaze (or at least, try out eye gaze!)

hmm as i understood it, the visual for head gaze and eye gaze in the unity editor is the same, so the pointer displayed when eye gaze simulation is enabled looks like the head gaze one. (You can't have both head and eye gaze pointer at the same time...)

Enabling eye simulation for Hololens2 profile seemed to be 'normal' to me. maybe the pointer for eye should display differently from head gaze to avoid confusion in editor.

maybe i don't get the real behavior of HL2, but eye tracking is always enable in the headset, or does teh head gaze is enabled by default and you can' switch' it to eye tracking somehow ? if eye tracking is always enable, i don't see why you would not want it to be simulated in editor .. 🤔 this may be the reason i'm still confused here.

the simulation profile should match the real hardware here, which was not the case if eye tracking is not enabled, and user may encounter a different behavior in the unity editor and when launching the app in the real hololens 2. => as long as it's the case, this is fine.
..and then i don't understand this :
> I think it may be good to add a script to validate that the default simulation profiles do not simulate eye tracking.

i get it that you want shell parity between headsets, but if their capabilities are different, this starts to be hard to get a common behavior, right ?
my initial **proposal was to have same behavior for head and eye pointers visibility rules** (after all they are both gaze pointers) but apparently this is not the wanted behavior (see #6320 discussion).

**Comment by gilbdev** [comment]
tldr : i don't think changing simulation profile would fix the #6496. 
what would happen in the real headset ?

**Comment by cre8ivepark** [comment]
@gilbdev HoloLens2 shell does not use any eye tracking features currently. The head gaze cursor shows up when the user say "Select". It is essentially hands free mode. When hand is visible again, gaze cursor disappears. There is no eye tracking component in this flow.

To use the eye tracking input, need to check 'Gaze Input' in VS and it displays the eye gaze cursor whenever hand is not visible. This is true with DefaultHoloLens2ConfigProfile.

**Comment by julenka** [comment]
>The fact that there is this side effect feels like a separate bug that we can fix. Whether or not that's critical determines whether or not we live with it, or roll it back and go again.

> I think my other general question is with respect to eye gaze and cursors, is it a true statement that the rules controlling eye gaze visibility and hands are the same ones that control head gaze visibility and hands? 

No, in fact @wiwei you modified the code yourself to have different rules for the gaze cursor when eye gaze are used instead of hands, have a look [here](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/27540471d9a27c37559ad4585356d3a213a5d1a5/Assets/MixedRealityToolkit.Services/InputSystem/GazePointerVisibilityStateMachine.cs#L46)

In particular, [this line](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/27540471d9a27c37559ad4585356d3a213a5d1a5/Assets/MixedRealityToolkit.Services/InputSystem/GazePointerVisibilityStateMachine.cs#L67) controls whether the gaze cursor should show up when eyes are active.

> If so we should just fix the underlying bug and maybe even leave this profile in because it gives developers the new eye tracking features on by default if they are targetting HL2 development

Perhaps the underlying bug is "when eye tracking is enabled on the default HoloLens 2 profile, eye gaze is used to to target objects instead of head gaze". Actually if you deploy & run hand interactions example scene with eye tracking enabled I think you do get the gaze cursor following the eyes right now, which is not actually matching the shell.

MRTK's current solution is as @crew8ivepark mentioned -- by default eye tracking is not enabled in visual studio, so we do not see the errant behavior.

@keveleigh  and @sostel you two have worked with the eye tracking in MRTK the most. It seems like currently there is no way to both enable eye tracking but still have the cursor behave like it does in the shell. 

It seems like the work that @gilbdev did actually did expose a larger bug in MRTK regarding eye input, thank you for this!

If we can fix the underlying issue (cursor does not match shell behavior when eye tracking is enabled) for 2.2, that would be great. If not, I propose we continue to complete this PR, since without reverting  #6393 we are getting a regression -- the cursor will behave incorrectly in the editor.

**Comment by julenka** [comment]
> Perhaps the underlying bug is "when eye tracking is enabled on the default HoloLens 2 profile, eye gaze is used to to target objects instead of head gaze". Actually if you deploy & run hand interactions example scene with eye tracking enabled I think you do get the gaze cursor following the eyes right now, which is not actually matching the shell.


I filed #6541 to track that bug

**Comment by julenka** [comment]
@wiwei and @davidkline-ms  thoughts on checking this in to revert the regression, and then fixing #6541 in 2.3?

**Comment by wiwei** [comment]
sgtm, I'm always a fan of rollback and fix right
